### PR TITLE
Delete code for NeevaMenu and overflow and cheatsheet flags

### PR DESCRIPTION
**Pull request by EdwardJTL**
- Remove code for neeva menu
  - SwiftUI views for the menu removed
  - Renamed button views
  - BrowserViewController
    - methods to present the menu removed
    - methods for handling menu functions relocated to overflow menu action
- Remove use of update tab overflow flag
- Remove use of cheatsheet query server flag
- Remove use of show neeva menu will move flag
  - Flag is kept in defaults in case we accidentally reused the same name and pick up the on device value
- Remove code presenting recipe banner 

TourManager is built around menu actions. Should we remove that too? @darinwf 

### How was this tested?
- [ ] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

Tests will break on this PR. need additional PRs to remove neeva menu tests and add them back as overflow tests

## Issues addressed
#2892

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -102,7 +102,8 @@ class BrowserViewController: UIViewController, ModalPresenter {
                         .environmentObject(self.gridModel.spaceCardModel)
                 }
             },
-            onNeevaMenuAction: self.perform(neevaMenuAction:))
+            onMenuAction:  { self.perform(overflowMenuAction: $0, targetButtonView: nil) }
```
Comment (darinwf) [inline Client/Frontend/Browser/BrowserViewController.swift:105]: nit: extra space before the `{` ... might need to run `swift-format.sh`

**Review by darinwf — APPROVED** [review]
LGTM!

**Comment by evancrow** [comment]
Please also remove the transition code for the Neeva Menu in the UITests. Under the `UITestBase+GoToExtension` file.

**Comment by EdwardJTL** [comment]
> Please also remove the transition code for the Neeva Menu in the UITests. Under the `UITestBase+GoToExtension` file.

All the test changes will be done in a separate PR
